plugins/destination/postgresql/client/insert.go

```diff
@@ -13,15 +13,32 @@ import (
 	"github.com/jackc/pgx/v5/pgconn"
 )
 
-// InsertBatch inserts records into the destination table. It forms part of the writer.MixedBatchWriter interface.
-func (c *Client) InsertBatch(ctx context.Context, messages message.WriteInserts) error {
+func (c *Client) pgTables(ctx context.Context) (map[string]struct{}, error) {
+	c.pgTablesToPKConstraintsMu.RLock()
 	// This happens when the CLI was invoked with `sync --no-migrate`
-	if c.pgTablesToPKConstraints == nil {
+	if len(c.pgTablesToPKConstraints) == 0 {
+		c.pgTablesToPKConstraintsMu.RUnlock()
```

Comment:
Why are we unlocking then locking again?

Reply:
because listTables requires an unlocked mutex